src/main/scala/awscala/dynamodbv2/DynamoDB.scala

```diff
@@ -340,13 +352,104 @@ trait DynamoDB extends aws.AmazonDynamoDB {
       .withLimit(limit)
       .withSegment(segment)
       .withTotalSegments(totalSegments)
+      .withConsistentRead(consistentRead)
+      .withReturnConsumedCapacity(aws.model.ReturnConsumedCapacity.TOTAL)
     if (!attributesToGet.isEmpty) {
       req.setAttributesToGet(attributesToGet.asJava)
     }
 
-    scan(req).getItems.asScala.map(i => Item(table, i)).toSeq
+    val pager = new ScanResultPager(table, scan(_), req, pageStatsCallback)
+    pager.toSeq // will return a Stream[Item]
   } catch { case e: aws.model.ResourceNotFoundException => Nil }
+}
+
+case class PageStats(page: Int, lastPage: Boolean, limit: Int, scanned: Int, items: Int, consumedCapacity: aws.model.ConsumedCapacity)
```

Comment:
Please separate source file except sealed trait and its sub types. (e.g.) move this case class to `PageStats.scala`.